### PR TITLE
Fix mypy errors in flytekit/types

### PR DESCRIPTION
**Pull request by aeioulisa**
# TL;DR
Fix all mypy errors in flytekit/types:
```
flytekit/types/pickle/pickle.py:35: error: Return type "Type[Any]" of "python_type" incompatible with return type "None" in supertype "FlytePickle"
flytekit/types/pickle/pickle.py:38: error: Incompatible return value type (got "Type[_SpecificFormatClass]", expected "Type[T]")
flytekit/types/file/file.py:242: error: Argument 1 to "get_format" of "FlyteFilePathTransformer" has incompatible type "Union[Type[FlyteFile[Any]], PathLike[Any]]"; expected "Type[FlyteFile[Any]]"
flytekit/types/file/file.py:345: error: Variable "expected_format" is not valid as a type
flytekit/types/file/file.py:345: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
flytekit/types/file/file.py:352: error: Type expected within [...]
flytekit/types/file/file.py:352: error: The type "Type[FlyteFile[Any]]" is not generic and not indexable
flytekit/types/directory/types.py:293: error: Variable "expected_format" is not valid as a type
flytekit/types/directory/types.py:293: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
flytekit/types/directory/types.py:303: error: Type expected within [...]
flytekit/types/directory/types.py:303: error: The type "Type[FlyteDirectory[Any]]" is not generic and not indexable
flytekit/types/file/__init__.py:31: error: Type expected within [...]
flytekit/types/file/__init__.py:31: error: Invalid type alias: expression is not a valid type
flytekit/types/file/__init__.py:31: error: The type "Type[FlyteFile[Any]]" is not generic and not indexable
flytekit/types/file/__init__.py:37: error: Type expected within [...]
flytekit/types/file/__init__.py:37: error: Invalid type alias: expression is not a valid type
flytekit/types/file/__init__.py:37: error: The type "Type[FlyteFile[Any]]" is not generic and not indexable
flytekit/types/file/__init__.py:43: error: Type expected within [...]
flytekit/types/file/__init__.py:43: error: Invalid type alias: expression is not a valid type
flytekit/types/file/__init__.py:43: error: The type "Type[FlyteFile[Any]]" is not generic and not indexable
flytekit/types/file/__init__.py:48: error: Type expected within [...]
flytekit/types/file/__init__.py:48: error: Invalid type alias: expression is not a valid type
flytekit/types/file/__init__.py:48: error: The type "Type[FlyteFile[Any]]" is not generic and not indexable
flytekit/types/file/__init__.py:54: error: Type expected within [...]
flytekit/types/file/__init__.py:54: error: Invalid type alias: expression is not a valid type
flytekit/types/file/__init__.py:54: error: The type "Type[FlyteFile[Any]]" is not generic and not indexable
flytekit/types/file/__init__.py:60: error: Type expected within [...]
flytekit/types/file/__init__.py:60: error: Invalid type alias: expression is not a valid type
flytekit/types/file/__init__.py:60: error: The type "Type[FlyteFile[Any]]" is not generic and not indexable
flytekit/types/file/__init__.py:66: error: Type expected within [...]
flytekit/types/file/__init__.py:66: error: Invalid type alias: expression is not a valid type
flytekit/types/file/__init__.py:66: error: The type "Type[FlyteFile[Any]]" is not generic and not indexable
flytekit/types/file/__init__.py:72: error: Type expected within [...]
flytekit/types/file/__init__.py:72: error: Invalid type alias: expression is not a valid type
flytekit/types/file/__init__.py:72: error: The type "Type[FlyteFile[Any]]" is not generic and not indexable
flytekit/types/file/__init__.py:77: error: Type expected within [...]
flytekit/types/file/__init__.py:77: error: Invalid type alias: expression is not a valid type
flytekit/types/file/__init__.py:77: error: The type "Type[FlyteFile[Any]]" is not generic and not indexable
flytekit/types/file/__init__.py:83: error: Type expected within [...]
flytekit/types/file/__init__.py:83: error: Invalid type alias: expression is not a valid type
flytekit/types/file/__init__.py:83: error: The type "Type[FlyteFile[Any]]" is not generic and not indexable
flytekit/types/directory/__init__.py:21: error: Type expected within [...]
flytekit/types/directory/__init__.py:21: error: Invalid type alias: expression is not a valid type
flytekit/types/directory/__init__.py:21: error: The type "Type[FlyteDirectory[Any]]" is not generic and not indexable
flytekit/types/schema/types.py:38: error: The return type of a generator function should be "Generator" or one of its supertypes
flytekit/types/schema/types.py:43: error: Free type variable expected in Generic[...]
flytekit/types/schema/types.py:73: error: Free type variable expected in Generic[...]
flytekit/types/schema/types.py:79: error: Argument 1 to "generate_ordered_files" has incompatible type "str"; expected "PathLike[Any]"
flytekit/types/schema/types.py:96: error: "SchemaReader" expects no type arguments, but 1 given
flytekit/types/schema/types.py:108: error: Argument 1 to "_read" of "LocalIOSchemaReader" has incompatible type "str"; expected "PathLike[Any]"
flytekit/types/schema/types.py:117: error: Argument 1 to "_read" of "LocalIOSchemaReader" has incompatible type "*List[str]"; expected "PathLike[Any]"
flytekit/types/schema/types.py:120: error: "SchemaWriter" expects no type arguments, but 1 given
flytekit/types/schema/types.py:122: error: Argument 2 to "__init__" of "SchemaWriter" has incompatible type "Optional[Dict[str, type]]"; expected "Dict[str, type]"
flytekit/types/schema/types.py:282: error: "None" not callable
flytekit/types/schema/types.py:285: error: Argument 1 to "SchemaWriter" has incompatible type "PathLike[Any]"; expected "str"
flytekit/types/schema/types.py:286: error: Argument 1 to "SchemaReader" has incompatible type "PathLike[Any]"; expected "str"
flytekit/types/schema/types.py:388: error: "dict" expects 2 type arguments, but 1 given
flytekit/types/schema/types.py:404: error: The type "Type[FlyteSchema]" is not generic and not indexable
flytekit/types/schema/types_pandas.py:22: error: Argument "columns" to "_read" of "ParquetIO" has incompatible type "Optional[List[str]]"; expected "List[str]"
flytekit/types/schema/types_pandas.py:89: error: "LocalIOSchemaReader" expects no type arguments, but 1 given
flytekit/types/schema/types_pandas.py:98: error: "LocalIOSchemaWriter" expects no type arguments, but 1 given
Found 59 errors in 7 files (checked 10 source files)
```

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Follow-up issue
https://github.com/lyft/flyte/issues/1330

### CODE REVIEW

File: flytekit/types/schema/types.py [hunk 1]
```diff
@@ -219,10 +222,10 @@ def format(cls) -> SchemaFormat:
 
     def __init__(
         self,
+        downloader: typing.Callable[[str, os.PathLike], None],
```
Comment (pingsutw) [inline flytekit/types/schema/types.py:225]: Could we keep the `downloader` default value as None?

Comment (aeioulisa) [reply flytekit/types/schema/types.py]: Sure, but "None" is not callable. Should I add some checks before using it?

Comment (pingsutw) [reply flytekit/types/schema/types.py]: yes, we need it

**Comment by wild-endeavor** [comment]
kinda lost as to why tests were failing, investigating

**Comment by kumare3** [comment]
Omg this is crazy. Awesome

**Comment by pingsutw** [comment]
Test errors seem unrelated to this PR. will rerun it after a few hours.
```
        if rsp.status_code != self._HTTP_OK:
            raise user.FlyteValueException(
                rsp.status_code,
>               "Request for data @ {} failed. Expected status code {}".format(from_path, type(self)._HTTP_OK),
            )
E           flytekit.common.exceptions.user.FlyteValueException: Value error!  Received: 406. Request for data @ https://cdn.sqlitetutorial.net/wp-content/uploads/2018/03/chinook.zip failed. Expected status code 200
```

**Review by wild-endeavor — APPROVED** [review]

**Pull request merged**